substrate/frame/salary/src/lib.rs

```diff
@@ -382,6 +391,66 @@ pub mod pallet {
 		}
 	}
 
+	#[pallet::tasks_experimental]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		#[pallet::task_list({
+			if let Some(status) = Status::<T, I>::get() {
+				let now = frame_system::Pallet::<T>::block_number();
+				let cycle_period = Pallet::<T, I>::cycle_period();
+				let result = if now >= status.cycle_start + cycle_period {
+					vec![()] // Success one task available: `()`.
+				} else {
+					vec![] // Failure no task available.
+				};
+				result.into_iter()
+			} else {
+				vec![].into_iter() // No task available.
+			}
+		})]
+		#[pallet::task_condition(|| {
+			let now = frame_system::Pallet::<T>::block_number();
+			let cycle_period = Pallet::<T, I>::cycle_period();
+			let Some(mut status) = Status::<T, I>::get() else { return false };
+			status.cycle_start.saturating_accrue(cycle_period);
+			now >= status.cycle_start
+		})]
+		#[pallet::task_weight(T::WeightInfo::bump_offchain())]
+		#[pallet::task_index(0)]
+		pub fn bump_offchain() -> DispatchResult {
+			let mut status = Status::<T, I>::get().ok_or(Error::<T, I>::NotStarted)?;
+			status.cycle_index.saturating_inc();
+			status.cycle_start = frame_system::Pallet::<T>::block_number();
+			status.budget = T::Budget::get();
+			status.total_registrations = Zero::zero();
+			status.total_unregistered_paid = Zero::zero();
+			Status::<T, I>::put(&status);
+
+			Pallet::deposit_event(Event::<T, I>::CycleStarted { index: status.cycle_index });
```

Comment:
I think we can factorize with the bump call into one function: `do_bump_unchecked`.